src/stratigraphy/elevation/elevation_extraction.py

```diff
@@ -0,0 +1,189 @@
+        extracted_elevation_informations.sort(key=lambda x: x.elevation, reverse=True)
+
+        # Return the first element of the sorted list
+        return extracted_elevation_informations[0] if extracted_elevation_informations else None
+
+    def get_elevation_from_lines(self, lines: list[TextLine], page: int) -> ElevationInformation:
+        r"""Matches the elevation in a string of text.
+
+        Args:
+            lines (list[TextLine]): Arbitrary string of text.
+            page (int): the page number (1-based) of the PDF document
+
+        Returns:
+            ElevationInformation: A list of potential elevation
+        """
+        matched_lines_rect = []
+
+        for line in lines:
+            text = self.preprocess(line.text)
+
+            # Check if the keyword line contains the elevation, extract it
+            elevation = extract_elevation(text)
+            if elevation:
+                # Pattern for matching depth (e.g., "1,48 m u.T.")
+                matched_lines_rect.append(line.rect)
+                break
+
+        # Get the union of all matched lines' rectangles
+        if len(matched_lines_rect) > 0:
+            # make sure the rectangles are unique - As some lines can contain both date and depth
+            unique_matched_lines_rect = []
+            for rect in matched_lines_rect:
+                if rect not in unique_matched_lines_rect:
+                    unique_matched_lines_rect.append(rect)
+
+            rect_array = np.array(unique_matched_lines_rect)
+            x0 = rect_array[:, 0].min()
+            y0 = rect_array[:, 1].min()
+            x1 = rect_array[:, 2].max()
+            y1 = rect_array[:, 3].max()
+            rect_union = fitz.Rect(x0, y0, x1, y1)
+        else:
+            rect_union = None
+
+        if elevation:
+            return ElevationInformation(elevation=elevation, rect=rect_union, page=page)
+        else:
+            raise ValueError("Could not extract all required information from the lines provided.")
+
+    def extract_elevation(self) -> ElevationInformation | None:
+        """Extracts the elevation information from a borehole profile.
+
+        Processes the borehole profile page by page and tries to find the feature key in the respective text of the
+        page.
+        """
+        for page in self.doc:
+            lines = extract_text_lines(page)
+            page_number = page.number + 1  # page.number is 0-based
+
+            found_feature_value = (
+                self.get_elevation_near_key(lines, page_number)
+                # or XXXX # Add other techniques here
+            )
+
+            if found_feature_value:
+                feature_value = getattr(found_feature_value, self.feature_name)
+                logger.info("Found %s on page %s: %s", self.feature_name, page_number, feature_value)
+                return found_feature_value
+
+        logger.info("No %s found in this borehole profile.", self.feature_name)
```

Comment:
this code no longer needs to be so generic, now that this method is back in the ElevationExtractor class, and no longer in the abstract parent class.